MoneyDataType/Currency.cs

```diff
@@ -0,0 +1,136 @@
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System;
+using System.Text.Json.Serialization;
+using OrchardCore.Commerce.Abstractions;
+using OrchardCore.Commerce.Serialization;
+
+namespace OrchardCore.Commerce.Money
+{
+    [JsonConverter(typeof(CurrencyConverter))]
+    [Newtonsoft.Json.JsonConverter(typeof(LegacyCurrencyConverter))]
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public struct Currency : ICurrency
+    {
+        internal readonly CultureInfo _culture;
+
+        public Currency(string cultureCode)
+            :this (CultureInfo.GetCultureInfo(cultureCode))
+        { }
+
+        internal Currency(CultureInfo culture)
+        {
+            if (culture is null)
+                throw new ArgumentNullException(nameof(culture));
+            if (culture.EnglishName.StartsWith("Unknown Locale") || culture.EnglishName.StartsWith("Invariant Language"))
+                throw new ArgumentOutOfRangeException(nameof(culture));
+
+            var region = new RegionInfo(culture.LCID);
+            if (region is null)
+                throw new ArgumentNullException(nameof(region));
+
+
+            EnglishName = region.CurrencyEnglishName;
+            NativeName = region.CurrencyNativeName;
+            Symbol = region.CurrencySymbol;
+            IsoCode = region.ISOCurrencySymbol;
+            DecimalPlaces = culture.NumberFormat.CurrencyDecimalDigits;
+            _culture = culture;
+        }
+
+        public Currency(string englishName, string nativeName, string symbol, string iSOSymbol, int decimalDigits = 2) 
+        {
+            EnglishName = englishName;
+            NativeName = nativeName;
+            Symbol = symbol;
+            IsoCode = iSOSymbol;
+            DecimalPlaces = decimalDigits;
+            _culture = null;
+        }
+
+        public string Culture => _culture?.Name??IsoCode;
```

Comment:
Invariant cultures do not have currencies.  Neither do neutral cultures.  The purpose of this change for me was to supply a list of known currencies as defined by NetStandard for use in many projects.  If you wish to provide additional currencies that are not related to those defined then I think they should be a separate package and made available via an additional currency provider.
Thoughts?

Reply:
It's the other way around conceptually for our use case: we're dealing with currencies here, some of which have cultures associated with them. That you can extract some useful data from .NET's built-in culture database is an implementation detail. If you want to make all reference to culture internal to this new library, actually I agree that would be better. It would then be OK to externalize the culture-neutral currencies out of the library into the main module.

Reply:
Culture changed to return CultureInfo instance.